### PR TITLE
Fixed typescript definition

### PR DESCRIPTION
**Pull request by GauravV-02**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [x] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

@jmca Fixed the second definition issue, got error for the first one if i made your suggested changes and  not sure about the third one

### CODE REVIEW

**Comment by jmca** [comment]
@GauravV-02 Regarding `Menu` component @:
https://github.com/ant-design/ant-design/blob/5f5986481e288cfd3be0ecacb263f2ef2f0be784/components/menu/index.tsx#L31
...and...
https://github.com/ant-design/ant-design/blob/5f5986481e288cfd3be0ecacb263f2ef2f0be784/components/menu/index.tsx#L64
Since `theme` has an inline union type (as opposed to a declared one with `type` keyword), I think you would need to cast as:
```
theme: 'light' as 'light',
```

Otherwise TS assumes `'light'` is a string and not the literal type of `'light'`.

**Comment by jmca** [comment]
@GauravV-02 Regarding @plrthink comment @ https://github.com/ant-design/ant-design/issues/8796#issuecomment-354921897

Glance at:
https://github.com/ant-design/ant-design/blob/5f5986481e288cfd3be0ecacb263f2ef2f0be784/components/popover/index.tsx#L14
...and...
https://github.com/ant-design/ant-design/blob/7fa05996957ef0eded21d810d71364d294ea947d/components/tooltip/index.tsx#L20

**Comment by GauravV-02** [comment]
@jmca Have a look

File: components/menu/index.tsx
Comment (jmca) [inline components/menu/index.tsx]: No double `as`, should be:
```theme: 'light' as 'light' | 'dark'```

Note to head repo maintainer:
I don't think there's a declared (alias) theme type of `'light' | 'dark'` (other than inline) in the current codebase. Might be beneficial to create one if it doesn't exist.

File: components/popover/index.tsx
Comment (jmca) [inline components/popover/index.tsx]: `TooltipPlacement` is a type alias so quotes are not needed.
Should be:
```
placement: 'top' as TooltipPlacement,
```

**Review by jmca — CHANGES_REQUESTED** [review]

File: components/popover/index.tsx
Comment (GauravV-02) [reply components/popover/index.tsx]: I am getting error, Cannot find TooltipPlacement

Comment (jmca) [reply components/popover/index.tsx]: @GauravV-02 You probably have to import that type from https://github.com/ant-design/ant-design/blob/7fa05996957ef0eded21d810d71364d294ea947d/components/tooltip/index.tsx#L10 

So...
```
import { TooltipPlacement } from '...filePath...'
```

Comment (GauravV-02) [reply components/popover/index.tsx]: I tried importing it before got error,  but this time it worked

**Review by jmca — APPROVED** [review]
@GauravV-02 Looks good to me.

**Comment by plrthink** [comment]
also LGTM 👍

File: components/popover/index.tsx
Comment (plrthink) [inline components/popover/index.tsx]: I think this import should be placed in the line 3, e.g. `import { AbstractTooltipProps, TooltipPlacement } from '../tooltip'`

File: components/popover/index.tsx
Comment (plrthink) [inline components/popover/index.tsx]: this one should also be `'hover' | 'focus' | 'click'` since the type it should have

**Comment by liamaharon** [comment]
This is still an issue for me and my team. @GauravV-02 @jmca @plrthink any chance to revisit?

**Comment by yesmeck** [comment]
@liamaharon Could you upload a reproduction repo?

**Comment by ghazalmalik13** [comment]
@yesmeck any chance to resolve this issue. basically we are facing this issue.
https://github.com/ant-design/ant-design/issues/8796

**Comment by yesmeck** [comment]
@ghazalmalik13 Could you create a reproduction repo?

**Comment by ghazalmalik13** [comment]
@liamaharon i created the repo. Please review.

**Comment by yesmeck** [comment]
@ghazalmalik13 Thanks for your reproduction!

**Comment by yesmeck** [comment]
Fixed in 921361313a3cc4c9bde471e7b42c0f5c6261aeab

**Comment by ghazalmalik13** [comment]
@yesmeck but it's not release than how we use this?

**Comment by yesmeck** [comment]
Will release this weekend.

**Comment by ghazalmalik13** [comment]
@yesmeck thanks, anxiously waiting for release.

**Comment by ghazalmalik13** [comment]
@yesmeck @zheeeng @GauravV-02  @jmca  this issue is still not merged.

**Comment by yesmeck** [comment]
@ghazalmalik13 Sorry, got busy last week, release this weekend.

**Pull request closed without merging**
